lib/Dialect/FIRRTL/Transforms/GrandCentralTaps.cpp

```diff
@@ -785,6 +809,40 @@ void GrandCentralTapsPass::processAnnotation(AnnotatedPort &portAnno,
   llvm_unreachable("portAnnos is never populated with unsupported annos");
 }
 
+StringAttr GrandCentralTapsPass::getOrAddInnerSym(Operation *op) {
+  auto attr = op->getAttrOfType<StringAttr>("inner_sym");
+  if (attr)
+    return attr;
+  auto module = op->getParentOfType<FModuleOp>();
+  auto name = getModuleNamespace(module).newName("gct_sym");
```

Comment:
I've seen this being used in other places for the base name.  E.g., `<passThatGeneratedMe_sym>`. I don't know if it's terrifically useful to include the pass, but it may be useful to include the name of the operation if it has one.  We've used stuff like `newName("__<name>__")`. This is perhaps too verbose, but may help output readability.  There are other options (like just using the name or `_name`).  We can sort this out after the PR lands.

Reply:
Yeah totally agreed. I need a similar pattern yet again in the `GrandCentralPass`, and there I've tried to just use the `"name"` attribute if present, which has the nice side-effect of the symbols usually matching the name.

That's the third time this whole `getOrAddInnerSym` pattern comes up... Once that's all in and working I'll take a look at factoring this out into a better place. Maybe even some op interface on declarations or so.